src/ssl.c

```diff
@@ -44405,13 +44405,35 @@ int wolfSSL_CTX_set1_curves_list(WOLFSSL_CTX* ctx, const char* names)
                                           (XSTRNCMP(name, "P-521", len) == 0)) {
             curve = WOLFSSL_ECC_SECP521R1;
         }
-        else if (XSTRNCMP(name, "X25519", len) == 0)
+        else if (XSTRNCMP(name, "X25519", len) == 0) {
             curve = WOLFSSL_ECC_X25519;
-        else if ((curve = wc_ecc_get_curve_id_from_name(name)) < 0)
+        }
+        else {
+            int ret = wc_ecc_get_curve_id_from_name(name);
+            if (ret < 0) {
+                return WOLFSSL_FAILURE;
+            }
+            curve = (word16)ret;
+        }
+
+        if (curve > (sizeof(word32) * WOLFSSL_BIT_SIZE)) {
```

Comment:
It looks like `curve` is `word16`. Should this be sizeof(word16) here?

Reply:
Using sizeof(word32) because of the size of ctx->disabledCurves. The static analysis report was complaining that later on when shifting 1 to the left more than the size of unsigned int size it was undefined behavior.